aea/crypto/fetchai.py

```diff
@@ -202,3 +202,36 @@ def is_transaction_settled(self, tx_digest: str) -> bool:
             # TODO: check the amount of the transaction is correct
             is_successful = True
         return is_successful
+
+    def validate_transaction(
+        self,
+        tx_digest: str,
+        seller: Address,
+        client: Address,
+        tx_nonce: str,
+        amount: int,
+    ) -> bool:
+        """
+        Check whether a transaction is valid or not.
+
+        :param seller: the address of the seller.
+        :param client: the address of the client.
+        :param tx_nonce: the transaction nonce.
+        :param amount: the amount we expect to get from the transaction.
+        :param tx_digest: the transaction digest.
+
+        :return: True if the random_message is equals to tx['input']
+        """
+
+        raise NotImplementedError
+
+    def generate_tx_nonce(self, seller: Address, client: Address) -> str:
+        """
+        Generate a random str message.
+
+        :param seller: the address of the seller.
+        :param client: the address of the client.
+        :return: return the hash in hex.
+        """
+
```

Comment:
the implementation can mirror the one from ethereum, use https://github.com/fetchai/ledger-api-py/blob/b000d20f1375f1b2b2b79ace3616efba18640020/fetchai/ledger/serialisation/sha256.py